packages/tc-markdown-parser/__tests__/property-value.spec.js

```diff
@@ -0,0 +1,330 @@
+
+		* chee-rabbits
+
+		## younger siblings
+
+		* ft-app-fruitcake
+	`);
+
+	expect(errors).toHaveLength(1);
+	expect(data.children).toEqual(['chee-rabbits']);
+	const [{ message }] = errors;
+	expect(message).toEqual(
+		'youngerSiblings is not permitted within markdown (to allow other people to edit it)',
+	);
+});
+
+test('properties with hasMany must be bulleted lists', async () => {
+	const { errors } = await parser.parseMarkdownString(here`
+		# name
+
+		## children
+
+		chee-rabbits
+	`);
+
+	expect(errors).toHaveLength(1);
+
+	const [{ message }] = errors;
+	expect(message).toContain('list');
+	expect(message).toContain('bullet');
+});
+
+test('properties without hasMany must not be bulleted lists', async () => {
+	const { errors } = await parser.parseMarkdownString(here`
+		# name
+
+		## some url
+
+		* https://ft.com
+	`);
+
+	expect(errors).toHaveLength(1);
+
+	const [{ message }] = errors;
+	expect(message).toContain('list');
+	expect(message).toContain('bullet');
+});
+
+test('subdocuments have their headers reduced two levels', async () => {
+	const { data } = await parser.parseMarkdownString(here`
+		# name
+
+		## some document
+
+		### hello
+	`);
+
+	expect(data).toEqual({
+		name: 'name',
+		someDocument: '# hello',
+	});
+});
+
+// There are currently no date fields permitted in runbook.md; lastServiceReviewDate is reserved for Ops manual entry
+test('date fields are coerced to iso strings', async () => {
+	const naiveJavaScriptIsoStringRegex = /^\d{4}(?:-\d{2}){2}T(?:\d{2}:){2}\d{2}\.\d{3}Z$/;
+	const { data } = await parser.parseMarkdownString(here`
+		# name
+
+		## some date
+
+		July 21 2018
+	`);
+
+	expect(data.someDate).toMatch(naiveJavaScriptIsoStringRegex);
+});
+
+test('date fields keep the correct date', async () => {
+	const { data } = await parser.parseMarkdownString(here`
+		# name
+
+		## some date
+
+		1965-09-17
+	`);
+
+	const date = new Date(data.someDate);
+
+	// gotta go fast
+	expect(date.getFullYear()).toBe(1965);
+	expect(date.getMonth()).toBe(8);
+	expect(date.getDate()).toBe(17);
+});
+
+test('date fields with bad dates are an error', async () => {
+	const { errors } = await parser.parseMarkdownString(here`
+		# name
+
+		## some date
+
+		mario's birthday
+	`);
+
+	expect(errors).toHaveLength(1);
+
+	const [{ message }] = errors;
+
+	expect(message).toContain("mario's birthday");
+	expect(message).toContain('Date');
+});
+
+test('explicit empty section should be an error', async () => {
+	const { data, errors } = await parser.parseMarkdownString(here`
+		# name
+
+		## some string
+
+		## another string
+		`);
+	expect(data).toEqual({
+		name: 'name',
+	});
+
+	expect(errors).toEqual([
+		{
+			line: 3,
+			message: 'property "someString" has no value',
+		},
+		{
+			line: 5,
+			message: 'property "anotherString" has no value',
+		},
+	]);
+});
+
+describe('dealing with html comments', () => {
+	it('html comments only in section should not be an error but be omitted in result', async () => {
```

Comment:
Is there a test somewhere that checks that the following is invalid?

```
			# name
			## some string
			
			## some document
			
```

Reply:
Just above tests https://github.com/Financial-Times/treecreeper/pull/93/files/acb8d01b66915104ad41817ec0dad960cfd0b383#diff-5624ce41fc406050be2a3489982935acR267-R289 but we lack the test about the document... I'll add